lib/client/indexer/proposals.ts

```diff
@@ -1,25 +1,234 @@
+                      proposer: new PublicKey(p.proposer_acct),
+                      slotEnqueued: p.initial_slot,
+                    },
+                    baseVaultAccount: {
+                      conditionalOnFinalizeTokenMint: new PublicKey(
+                        passMarket?.base_mint_acct ?? 5
+                      ),
+                      conditionalOnRevertTokenMint: new PublicKey(
+                        failMarket?.base_mint_acct ?? 5
+                      ),
+                      protocol: relatedProtocol,
+                      // TODO adding this to get the types to work, but this is not long term
+                      settlementAuthority: new PublicKey(4),
+                      // TODO use conditional vault when it's available, deposit/withdrawal won't work until then
+                      underlyingTokenAccount: new PublicKey(
+                        relatedDao.baseToken.publicKey ?? 5
+                      ),
+                      underlyingTokenMint: new PublicKey(
+                        relatedDao.baseToken.publicKey ?? 5
+                      ),
+                    },
+                    quoteVaultAccount: {
+                      conditionalOnFinalizeTokenMint: new PublicKey(
+                        passMarket?.quote_mint_acct ?? 5
+                      ),
+                      conditionalOnRevertTokenMint: new PublicKey(
+                        passMarket?.quote_mint_acct ?? 5
+                      ),
+                      protocol: relatedProtocol,
+                      // TODO adding this to get the types to work, but this is not long term
+                      settlementAuthority: new PublicKey(4),
+                      underlyingTokenAccount: new PublicKey(
+                        relatedDao.quoteToken.publicKey ?? 5
+                      ),
+                      underlyingTokenMint: new PublicKey(
+                        relatedDao.quoteToken.publicKey ?? 5
+                      ),
+                    },
+                    content: p.proposal_details[0].content ?? "",
+                    description: p.proposal_details[0].description ?? "",
+                    // TODO figure this out by slot enqueued maybe
+                    creationDate: p.initial_slot,
+                    dao: {
+                      publicKey: new PublicKey(d.dao_acct),
+                      daoAccount: {
+                        proposalCount: d.proposals.length,
+                        treasury: new PublicKey(d.treasury_acct ?? 5),
+                        usdcMint: new PublicKey(
+                          d.tokenByQuoteAcct?.mint_acct ?? 5
+                        ),
+                        tokenMint: new PublicKey(
+                          d.tokenByBaseAcct?.mint_acct ?? 5
+                        ),
+                      },
+                    },
+                    // TODO figure this out by slot enqueued maybe
+                    finalizationDate: new Date(new Date().getDate() + 20),
```

Comment:
Yea this is we can get by adding in the regime for the program / proposals.. Eg 3 days of slots or 5 days of slots...

Reply:
But slot to date is gonna be weird, may want to index something or write a helper..

Reply:
So if we index it then another field? If that makes the UI avoid heavy lifting it would be good

Reply:
The thing is, slot to date time is useful, so an entire table or some slot to date check every 10-20 minutes so you reference from there...